src/module.c

```diff
@@ -6704,6 +6733,115 @@ int RM_SetModuleUserACL(RedisModuleUser *user, const char* acl) {
     return ACLSetUser(user->user, acl, -1);
 }
 
+/* Creates a user ID from the user created through the redis module interface.
+ * The user ID can be used later, in order to check if command, key or channel
+ * can be executed or accessed according to the ACLs rules associated with that user.
+ * The caller can later free the user ID using the function RM_FreeModuleUserID
+ *
+ * On success a RedisModuleUserID object is returned, otherwise
+ * NULL is returned and errno is set to ENOENT. */
+ RedisModuleUserID *RM_CreateModuleUserID(RedisModuleCtx *ctx) {
```

Comment:
```suggestion
/* Retrieve the user ID of the client connection behind the current context.
 * The user ID can be used later, in order to check if command, key or channel
 * can be executed or accessed according to the ACLs rules associated with that user.
 * The caller can later free the user ID using the function RM_FreeUserID
 *
 * On success a RedisModuleUserID object is returned, otherwise
 * NULL is returned. */
 RedisModuleUserID *RM_GetCurrentUserID(RedisModuleCtx *ctx) {
```

Reply:
@yossigo I agree that `RM_GetCurrentUserID` is a better name, but using the "Create" word implies that it is not just a reference for exiting object, but we created it, and the user should be responsible for freeing it, WDYT?

Reply:
@YaacovHazan C being a rather low level language, I think it's reasonable to assume you need to release something you've received as a result of a `get`. Some examples:

* `getaddrinfo()` -> `freeaddrinfo()`
* `getipnodebyname()` -> `freehostent()`

Also, the Redis Module API already has similar use cases - just to name a few:
* `RM_GetClientUserNameById`
* `RM_ServerInfoGetField`